### PR TITLE
refactor(Grid.js): replace eval() with template literals and other improvements

### PR DESCRIPTION
**Pull request by ASchmidt1024**
- Introduce async/await to the functions that return promises
- Replace eval() with template literals and expressions inside ${}
- Tiny improvements to increase readability

### CODE REVIEW

File: src/es/components/organisms/grid/Grid.js [hunk 1]
```diff
@@ -12,87 +12,94 @@ import { Shadow } from '../../prototypes/Shadow.js'
  * @css {}
  */
 export default class Grid extends Shadow() {
-  connectedCallback () {
-    this.hidden = true
-    const showPromises = []
-    if (this.shouldComponentRenderCSS()) showPromises.push(this.renderCSS())
-    if (this.shouldComponentRenderHTML()) showPromises.push(this.renderHTML())
-    Promise.all(showPromises).then(() => (this.hidden = false))
+  async connectedCallback() {
```
Comment (Weedshaker) [inline src/es/components/organisms/grid/Grid.js:15]: should be done for all components... in out tide up schedule.

File: src/es/components/organisms/grid/Grid.js [hunk 2]
```diff
@@ -12,87 +12,94 @@ import { Shadow } from '../../prototypes/Shadow.js'
...
 
   /**
-   * evaluates if a render is necessary
+   * Evaluates whether a render is necessary.
    *
    * @return {boolean}
    */
-  shouldComponentRenderCSS () {
-    return !this.root.querySelector(`:host > style[_css], ${this.tagName} > style[_css]`)
+  shouldRenderCSS() {
```
Comment (Weedshaker) [inline src/es/components/organisms/grid/Grid.js:33]: totally agree, but if we change this functions name, we gotta change it through the whole solution, to keep the namespaces within the components consistent. this for the scenario of inheritance.

File: src/es/components/organisms/grid/Grid.js [hunk 3]
```diff
@@ -12,87 +12,94 @@ import { Shadow } from '../../prototypes/Shadow.js'
...
   /**
-   * evaluates if a render is necessary
+   * Evaluates whether a render is necessary.
    *
    * @return {boolean}
    */
-  shouldComponentRenderHTML () {
-    return !this.grid
+  shouldRenderHTML() {
+    return !this.grid;
```
Comment (Weedshaker) [inline src/es/components/organisms/grid/Grid.js:43]: run the linter, and it is going to remove your semicolons LOL

File: src/es/components/organisms/grid/Grid.js [hunk 4]
```diff
@@ -12,87 +12,94 @@ import { Shadow } from '../../prototypes/Shadow.js'
...
   /**
-   * renders the o-grid css
+   * Renders the o-grid CSS.
    *
    * @return {Promise<void>}
    */
-  renderCSS () {
-    this.css = /* css */`
+  async renderCSS() {
+    const url = import.meta.url.replace(/(.*\/)(.*)$/, '$1');
```
Comment (Weedshaker) [inline src/es/components/organisms/grid/Grid.js:52]: good catch! we have a scoped variable "this.importMetaUrl" on shadow, but it has to be fed with the import.meta.url through the constructor. this is part of the tide up sprint. see: https://github.com/mits-gossau/web-components-toolbox/blob/master/README.md#mater-branch-locked- transform all import.meta.url.replace(/(./)(.)$/, '$1') to this.importMetaUrl

File: src/es/components/organisms/grid/Grid.js [hunk 5]
```diff
@@ -12,87 +12,94 @@ import { Shadow } from '../../prototypes/Shadow.js'
...
+    this.css = /* css */ `
       :host > section {
-        display:grid;
-        ${this.hasAttribute('height')
-          ? `height: ${this.getAttribute('height') || '100%'};`
-          : ''
-        }
+        display: grid;
+        ${this.hasAttribute('height') ? `height: ${this.getAttribute('height') || '100%'};` : ''}
```
Comment (Weedshaker) [inline src/es/components/organisms/grid/Grid.js:56]: our linter standardjs loves to make linebreaks before ? and : conditions

File: src/es/components/organisms/grid/Grid.js (hunk 2 above)
Comment (ASchmidt1024) [reply src/es/components/organisms/grid/Grid.js]: Well, than we should do it at once. I'll roll back.

File: src/es/components/organisms/grid/Grid.js (hunk 4 above)
Comment (ASchmidt1024) [reply src/es/components/organisms/grid/Grid.js]: If okay, I would do it one by one.

**Pull request closed without merging**